src/plugins/workspace/public/utils.ts

```diff
@@ -260,6 +260,24 @@ export const convertNavGroupToWorkspaceUseCase = ({
   order,
 });
 
+const compareFeatures = (
+  features1: WorkspaceUseCaseFeature[],
+  features2: WorkspaceUseCaseFeature[]
+) => {
+  const featureSerialize = ({ id, title }: WorkspaceUseCaseFeature) => `${id}-${title}`;
+  const features1Set = new Set(features1.map(featureSerialize));
+  const features2Set = new Set(features2.map(featureSerialize));
+  if (features1Set.size !== features2Set.size) {
+    return false;
+  }
+  for (const feature of features1Set) {
+    if (!features2Set.has(feature)) {
+      return false;
+    }
+  }
+  return true;
+};
+
```

Comment:
Perhaps we can simplify the compare by sorting the serialized array and see if JSON.stringify(sortArray) equals?